onnxruntime/core/providers/rocm/shared_inc/fpgeneric.h

```diff
@@ -0,0 +1,604 @@
+                                                       const double* A, int lda,
+                                                       long long int strideA,
+                                                       const double* B, int ldb,
+                                                       long long int strideB,
+                                                       const double* beta,
+                                                       double* C, int ldc,
+                                                       long long int strideC,
+                                                       int batchCount){
+  return rocblas_dgemm_strided_batched(handle, transa, transb, m, n, k, alpha, A, lda, strideA, B, ldb, strideB, beta, C, ldc, strideC, batchCount);
+}
+
+inline rocblas_status rocblasGemmStridedBatchedHelper(rocblas_handle handle,
+                                                       rocblas_operation transa,
+                                                       rocblas_operation transb,
+                                                       int m, int n, int k,
+                                                       const __half* alpha,
+                                                       const __half* A, int lda,
+                                                       long long int strideA,
+                                                       const __half* B, int ldb,
+                                                       long long int strideB,
+                                                       const __half* beta,
+                                                       __half* C, int ldc,
+                                                       long long int strideC,
+                                                       int batchCount) {
+  float h_a = onnxruntime::math::halfToFloat(*reinterpret_cast<const uint16_t*>(alpha));
+  float h_b = onnxruntime::math::halfToFloat(*reinterpret_cast<const uint16_t*>(beta));
+  return rocblas_gemm_strided_batched_ex(handle,
+                                         transa,
+                                         transb,
+                                         m, n, k,
+                                         &h_a,
+                                         A, rocblas_datatype_f16_r, lda, strideA,
+                                         B, rocblas_datatype_f16_r, ldb, strideB,
+                                         &h_b,
+                                         C, rocblas_datatype_f16_r, ldc, strideC,
+                                         C, rocblas_datatype_f16_r, ldc, strideC,
+                                         batchCount,
+                                         rocblas_datatype_f32_r,
+                                         rocblas_gemm_algo_standard, 0, 0);
+}
+
+// // axpy
+// inline hipblasStatus_t hipblasAxpyHelper(hipblasHandle_t handle, int n, const float* alpha, const float* x, int incx, float* y, int incy) {
```

Comment:
Is this an example of unused code to be cleaned up in CUDA as well? Or is it commented for another reason?

Reply:
yes.